### ISSUE TITLE
Make available on npm

### ISSUE THREAD

**Issue by chadoh** [picnicss/picnic#53]
Hello! I have a project where I'm not using bower, just straight npm for both server and browser dependencies. (I'm new to this whole world, maybe people will be upset that that's not how I'm _supposed to do it_. Whatever. I started with someone else's project, and I rather like my reduced number of dependencies.)

It would be nice to have access to this on npmjs.com. Do you mind publishing it there?

**Comment by franciscop** [comment]
Hi @chadoh. I have been considering the implications of this issue. Are you using npm to manage any other (preferably [s]css) front-end dependency?

The main issue is that you cannot just do `var picnic = require('picnic');` as you would normally do with any npm module. The files would be loaded under `node_modules/picnic/src/picnic.scss`. Do you control-version that folder? Because [some people do and some don't](http://stackoverflow.com/q/11459475/938236). Also your workflow would depend on gulp, grunt or command lines that you use.

However in the plus side this doesn't differ so much from bower and it should be pretty much trivial to maintain.

**Comment by chadoh** [comment]
Hello! I saw this done with Bootstrap in [this starter app](https://github.com/auth0/react-flux-jwt-authentication-sample/blob/gh-pages/package.json). It uses [rework](https://github.com/auth0/react-flux-jwt-authentication-sample/blob/gh-pages/package.json#L52) and a couple related tools to [include bootstrap](https://github.com/auth0/react-flux-jwt-authentication-sample/blob/gh-pages/index.css#L1) right from npm. Pretty clean stuff!

**Comment by chadoh** [comment]
What do you think? It would be really nice to have this for my project. Can I help somehow?

**Comment by franciscop** [comment]
I am sorry I didn't have time previously, I will attempt to do it this weekend, I think this is the relevant package from rework:

https://github.com/reworkcss/rework-npm

**Comment by chadoh** [comment]
I'm now switching to Webpack, and it also encourages me to [prefer npm over bower](https://webpack.github.io/docs/usage-with-bower.html).

**Comment by chadoh** [comment]
Oh, and LOTS of people use webpack to manage their [S]CSS. It seems like it may be becoming [the standard way of doing frontend coding](http://chrispearce.co/elephants-the-three-code-ilities-two-months-with-css-modules/), at least for a few months! :stuck_out_tongue:

**Comment by franciscop** [comment]
Okay I've been trying to clean it up to publish it, but since there's the website mixed up with the library there's a bit of a problem. Working on it (:

**Comment by franciscop** [comment]
I've contacted the creator of [this picnic](https://www.npmjs.com/package/picnic) in NPM to see if he's planning to keep it alive (as the github seems dead). I'll update when I get an answer.

However I'm not completely sure if I'm doing it right, you just want me to publish it in npm? Or should I install some packages to do some magic so you get it ready? I'm unsure if the magic happens on picnic's end or on your setup's end.

**Comment by chadoh** [comment]
I'm not totally sure how things are packaged for npm, either—[this page seems like it has lots of good info](https://webpack.github.io/docs/usage-with-bower.html#prefer-modules-from-npm-over-bower), though.

Excerpt:

> Prefer modules from npm over bower
> ----------------------
> 
> In many cases modules from npm are better than the same module from bower. Bower mostly contain only concatenated/bundled files which are:
> 
> * More difficult to handle for webpack
> * More difficult to optimize for webpack
> * Sometimes only useable without a module system
> 
> So prefer to use the CommonJs-style module and let webpack build it.

**Comment by franciscop** [comment]
I read that bit, however the CommonJs seems to be a tool for pure JS. I just received the answer *Yes sure you can use it, I no longer maintain that package* so I might get it ready sooner than expected

**Comment by chadoh** [comment]
I recently used [this library](https://github.com/gilbarbara/react-joyride), which [instructs](https://github.com/gilbarbara/react-joyride#styles) users to include some CSS from their `node_modules` directory. Maybe that will help?

**Comment by franciscop** [comment]
That's basically how I was thinking on doing it, but I think it's a really dirty way since you are manually poking into the library and expecting file structure to remain unchanged...

**Comment by chadoh** [comment]
Yeah, hmm... [Bootstrap](http://getbootstrap.com/getting-started/#download-npm) seems like they're probably doing it in a better way, using some sort of setting in package.json.

**Comment by franciscop** [comment]
Okay that makes a lot more of sense:

```json
{
  "style": "dist/css/bootstrap.css",
  "less": "less/bootstrap.less",
}
```

This way the author of the package can keep some control over the routes and whether they want to change them. I can surely do something like this

**Comment by fleeting** [comment]
Just wanted to note real quick that you can use `npm` to install any repo on GitHub that has a valid `./package.json`. For Picnic you can do `npm install https://github.com/picnicss/picnic.git` and it will install it to `./node_modules` as expected. It doesn't have to be published to npm to allow that. Of course having it published to npm is a plus for easier install but not required if looking for a quick work around.

**Comment by franciscop** [comment]
Apparently I just published it as `picnic` (version 4.1.2), I'll check that it works tomorrow (:

**Comment by franciscop** [comment]
[Check it out](https://www.npmjs.com/package/picnic)

**Comment by chadoh** [comment]
Cool! It worked! You may want to include instructions in the README for  how to use it with npm + webpack. Here's what I did:

``` bash
npm i -S picnic
```

Then in my main css file, at the top:

``` css
@import "../node_modules/picnic/releases/picnic.css";
```

Here's [my code diff](https://github.com/chadoh/life/commit/82ebfb7e18ffa1afa28dd727715207f4f3e29571) – it might not be open source forever, but for now it is!

**Comment by franciscop** [comment]
That's awesome, I'm glad to hear it, I'l publish it right away. Just curious, do you know any advantage of using npm vs bower besides keeping your current workflow?

**Comment by chadoh** [comment]
I like it mostly because it's one fewer tool, but then there are also all of the benefits cited by webpack, which I mentioned in this thread [previously](https://github.com/picnicss/picnic/issues/53#issuecomment-160740807).

**Comment by franciscop** [comment]
Ohh I see, sorry I didn't see that Excerpt section. If you need anything else or have any other question about Picnic please ask (:

**Comment by chadoh** [comment]
Update on how I had to include it: because of some very sensible production bundling in webpack that I set up [per the guidance of SurviveJS](http://survivejs.com/webpack_react/building_kanban/#defining-a-vendor-entry-point), `picnic` can't be in the `dependencies` section of `package.json`. Instead, it should be in `devDependencies`. 

My earlier instructions should be updated to:

``` bash
npm i -D picnic
```

and then the same from there.
